File: backend/src/main/kotlin.nikolay/otdo/posts/PostsService.kt
package otdo.posts

import org.springframework.stereotype.Service
import otdo.posts.model.Post
import otdo.posts.model.PostType

@Service
class PostsService(private val servicesRepository: PostsRepository) {
    fun getPosts(postType: PostType): List<Post> {
        return servicesRepository.getPosts(postType)
    }
}

File: frontend/src/app/pages/main/posts.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Post, PostType} from "./model/Post";
import {environment} from "../../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class PostsService {
  private readonly domain: string;

  constructor(private http: HttpClient) {
    this.domain = environment._SERVER + '/posts';
  }

  getPosts(postType: PostType): Observable<Post[]> {
    return this.http.get<Post[]>(`${this.domain}/${postType}`);
  }
}
